### ISSUE TITLE
The code in "Training a Text Classification Model with Transformer" seems incorrect

### ISSUE THREAD

**Issue by sankaran45** [flairNLP/flair#2284]
https://github.com/flairNLP/flair/blob/master/resources/docs/TUTORIAL_7_TRAINING_A_MODEL.md

My student tried this code and compared it against a vanilla Flair + DocumentLSTMEmbedding model. The Transformer code surprisingly gave poor results. We find that the learning rate hardly changes from 3E-5 probably because of the default anneal on plateau scheduler. Normally, we will expect learning rate to get annealed at the end of each batch so that by the end of 3 epochs it has reduced significantly to a small value (which i think is how HuggingFace transformer library examples work).

How can I get a similar behavior ? I can use a different StepLR scheduler - but these schedulers require the .step() function to be invoked at the end of each batch.
